src/components/core/DpDataTable/DpDataTable.vue

```diff
@@ -375,8 +375,20 @@ export default {
       const firstRow = this.tableEl.firstChild
       const tableHeaders = Array.prototype.slice.call(firstRow.childNodes)
       tableHeaders.forEach(tableHeader => {
-        const width = tableHeader.getBoundingClientRect().width
-        tableHeader.style.width = width + 'px'
+        /**
+         * The convert of the component DpTableHeader.vue from a functional component to a template indicates,
+         * that the DOM Element structure changed, so that the tableHeaders doesnt just contains HTML elements,
+         * it contains other code parts which are not HTML elements.
+         * So you get an error, that the method getBoundingClientRect is not a function,
+         * because it needs the same dom structure as before and the belonging HTML elements.
+         * So here needs to be a check if the tableHeader.nodeName is a 'th' element for this case.
+         * A general check if the tableHeader.nodeName is an HTML element here is inconvenient,
+         * so here its okay to do the check with directly with the 'th' element.
+         */
+        if(tableHeader.nodeName === "TH") {
```

Comment:
Are You sure You have to check if its a TH? Isn't it enough to check if its a node?
`if(tableHeader.nodeName)` ?

And also its good to know where the bug came from right now, I think the depth of detail don't helps anyone in the feature. I suggest to shorten the comment.

And aside from that. Did You find out why There are none-TH elements in the Array?

Reply:
replacing `node` instead of `nodeName` didnt work. But it works with `nodeTyoe`, so I changed the check this way. done https://github.com/demos-europe/demosplan-ui/commit/9c36b6e1589b293dbd4b256ffa0880f977a8b045

Reply:
the reason why there are other elements are because of the v-if structure in the template. see comment in the code